// ngx-bottom-nav/esm2015/lib/bottom-nav/bottom-nav.component.js
import { Component, HostBinding, Input } from '@angular/core';
/**
 * This component is a [Material-style bottom navigation bar]{@link https://material.io/design/components/bottom-navigation.html}.
 * It receives an array of [BottomNavItems]{@link BottomNavItem} in order to render navigation buttons.
 * With [Material theming]{@link https://github.com/latusinski/ngx-drawer-layout#enabling-material-theming}
 * enabled, this component changes its style, when the {@link color} property.
 */
export class BottomNavComponent {
    constructor() {
        /**
         * Navigation items, which are rendered inside the navigation bar.
         */
        this.items = [];
    }
    get colorPrimary() {
        return this.color === 'primary';
    }
    get accentPrimary() {
        return this.color === 'accent';
    }
    get warnPrimary() {
        return this.color === 'warn';
    }
    /**
     * Returns true if the length of {@link items} is greater than 4, which hides labels on inactive items.
     */
    get hideLabels() {
        return this.items.length > 4;
    }
}
BottomNavComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-bottom-nav',
                template: "<mat-toolbar [color]=\"color\">\n  <button\n    ngx-bottom-nav\n    *ngFor=\"let i of items\"\n    [hideLabel]=\"hideLabels\"\n    [routerLink]=\"i.routerLink\"\n    [label]=\"i.label\"\n    [exact]=\"i.exact\"\n  >\n    <mat-icon ngxBottomNavIcon>{{ i.icon }}</mat-icon>\n  </button>\n  <ng-content select=\"[ngx-bottom-nav]\"></ng-content>\n</mat-toolbar>\n",
                styles: [":host{--ngx-bottom-nav-background-color:transparent;display:block;width:100%}mat-toolbar{background:var(--ngx-bottom-nav-background-color);display:flex;height:56px;justify-content:center;overflow:hidden;padding:0}mat-toolbar ngx-bottom-nav-button{flex:1 1 auto;height:100%}"]
            },] }
];
BottomNavComponent.propDecorators = {
    items: [{ type: Input }],
    color: [{ type: Input }],
    colorPrimary: [{ type: HostBinding, args: ['class.color-primary',] }],
    accentPrimary: [{ type: HostBinding, args: ['class.color-accent',] }],
    warnPrimary: [{ type: HostBinding, args: ['class.color-warn',] }]
};
