// ngx-bottom-nav/fesm2015/ngx-bottom-nav.js
import { Component, Input, HostBinding, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatRippleModule } from '@angular/material/core';
import { MatIconModule } from '@angular/material/icon';
import { MatToolbarModule } from '@angular/material/toolbar';
import { RouterModule } from '@angular/router';

/**
 * This component is a [Material-style bottom navigation bar]{@link https://material.io/design/components/bottom-navigation.html}.
 * It receives an array of [BottomNavItems]{@link BottomNavItem} in order to render navigation buttons.
 * With [Material theming]{@link https://github.com/latusinski/ngx-drawer-layout#enabling-material-theming}
 * enabled, this component changes its style, when the {@link color} property.
 */
class BottomNavComponent {
    constructor() {
        /**
         * Navigation items, which are rendered inside the navigation bar.
         */
        this.items = [];
    }
    get colorPrimary() {
        return this.color === 'primary';
    }
    get accentPrimary() {
        return this.color === 'accent';
    }
    get warnPrimary() {
        return this.color === 'warn';
    }
    /**
     * Returns true if the length of {@link items} is greater than 4, which hides labels on inactive items.
     */
    get hideLabels() {
        return this.items.length > 4;
    }
}
BottomNavComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-bottom-nav',
                template: "<mat-toolbar [color]=\"color\">\n  <button\n    ngx-bottom-nav\n    *ngFor=\"let i of items\"\n    [hideLabel]=\"hideLabels\"\n    [routerLink]=\"i.routerLink\"\n    [label]=\"i.label\"\n    [exact]=\"i.exact\"\n  >\n    <mat-icon ngxBottomNavIcon>{{ i.icon }}</mat-icon>\n  </button>\n  <ng-content select=\"[ngx-bottom-nav]\"></ng-content>\n</mat-toolbar>\n",
                styles: [":host{--ngx-bottom-nav-background-color:transparent;display:block;width:100%}mat-toolbar{background:var(--ngx-bottom-nav-background-color);display:flex;height:56px;justify-content:center;overflow:hidden;padding:0}mat-toolbar ngx-bottom-nav-button{flex:1 1 auto;height:100%}"]
            },] }
];
BottomNavComponent.propDecorators = {
    items: [{ type: Input }],
    color: [{ type: Input }],
    colorPrimary: [{ type: HostBinding, args: ['class.color-primary',] }],
    accentPrimary: [{ type: HostBinding, args: ['class.color-accent',] }],
    warnPrimary: [{ type: HostBinding, args: ['class.color-warn',] }]
};

class BottomNavButtonComponent {
}
BottomNavButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'button[ngx-bottom-nav]',
                template: "<div\n  class=\"ngx-bottom-nav-button\"\n  [routerLink]=\"routerLink\"\n  routerLinkActive=\"active\"\n  [routerLinkActiveOptions]=\"{ exact: exact }\"\n  [class.label-hidden]=\"hideLabel\"\n>\n  <ng-content select=\"[ngxBottomNavIcon]\"></ng-content>\n  <div class=\"label\">{{ label }}</div>\n</div>\n",
                styles: [":host{--ngx-bottom-nav-button-active-color:#000;--ngx-bottom-nav-button-color:#646464;-moz-user-select:none;-webkit-tap-highlight-color:transparent;-webkit-user-select:none;align-items:center;background:transparent;border:none;color:inherit;cursor:pointer;display:flex;flex-direction:column;height:calc(100% - 20px);justify-content:center;max-width:168px;min-width:80px;padding:8px 12px 12px;text-decoration:none;user-select:none}.ngx-bottom-nav-button:focus,:host{outline:none}.ngx-bottom-nav-button:not(.active){color:var(--ngx-bottom-nav-button-color)}.ngx-bottom-nav-button.active{--ngx-bottom-nav-button-color:var(--ngx-bottom-nav-button-active-color)}:host(.active:not(.label-hidden)){padding-top:7px}.label{font-size:12px;line-height:12px}:host(.active) .label{font-size:13px;line-height:13px}:host(.label-hidden:not(.active)){height:calc(100% - 32px);padding-bottom:16px;padding-top:16px}:host(.label-hidden:not(.active)) .label{display:none}"]
            },] }
];
BottomNavButtonComponent.propDecorators = {
    hideLabel: [{ type: Input }],
    icon: [{ type: Input }],
    label: [{ type: Input }],
    routerLink: [{ type: Input }],
    exact: [{ type: Input }]
};

class BottomNavModule {
}
BottomNavModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    BottomNavComponent,
                    BottomNavButtonComponent,
                ],
                imports: [
                    CommonModule,
                    MatIconModule,
                    MatRippleModule,
                    MatToolbarModule,
                    RouterModule
                ],
                exports: [
                    BottomNavComponent,
                    BottomNavButtonComponent
                ]
            },] }
];

/*
 * Public API Surface of bottom-nav
 */

/**
 * Generated bundle index. Do not edit.
 */

export { BottomNavButtonComponent, BottomNavComponent, BottomNavModule };
